### DISCUSSION TITLE
Help with making Alire available on Homebrew

### DISCUSSION THREAD

**Discussion by Joebeazelman** [Homebrew/discussions#4657] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

Alire is a package and build management utility for Ada development. It works across Windows, macOS, and other UNIX environments and is installable as a single executable download.  In order to introduce more developers to Ada's powerful capabilities, accessibility via Homebrew's main tap is vital. It's in our mutual interest to expand the choices of developer tools and languages.

Jesper Quorning volunteered to complete this task but reached an impasse after submitting a pull request to Homebrew (see [#117527](https://github.com/Homebrew/homebrew-cask/pull/117527)). The process of building Alire from the source is a complex interaction of tools and compilers and is very brittle outside of a highly controlled build environment. Realizing the weight of this effort, Jesper couldn't commit to completing the task. Since then, I've volunteered to take over.  

Fortunately, Alire's developers did an exquisite job on the build process and thoroughly documented it. The simplest way to build Alr from the source is to use Homebrew to download the [GCC compiler collection](https://formulae.brew.sh/formula/gcc), clone Alire's source from Github, set up the necessary environment variables, and execute the build scripts. I'm hoping we don't have to build the Ada compiler by leveraging the existing GCC formula.

Please advise if this approach fulfills your build-from-source requirement. Also, please provide me with any resources which detail the process. Wading through the massive amount of information to determine whether it's applicable is difficult.

**Comment by osalbahr** [comment]
I hope https://docs.brew.sh/Formula-Cookbook answers some of your questions.

**Comment by tlk** [comment]
Check out https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap#creating-a-tap - if you create a "custom tap", your users can install your software like this `brew install xyz/repo/yoursoftware` where xyz is your github username. You may be able to use the work as-is from https://github.com/Homebrew/homebrew-cask/pull/117527.

**Reply by Joebeazelman** [reply]
I want this to be an official release so anyone can find it.

**Comment by carlocab** [comment]
These PRs might be a useful reference:

- https://github.com/Homebrew/homebrew-core/pull/77641
- https://github.com/Homebrew/homebrew-core/pull/77402

I suggest starting from Homebrew/homebrew-core#77641, as that doesn't seem too far from getting into a state that can be merged.

**Reply by Joebeazelman** [reply]
I'm lost here. I'm assuming these PRs are for Ada compiler support, correct? If I have to build Alire, it will need a fairly recent Ada compiler. I'm assuming these PRs are for building and installing the Ada compiler, correct?  That said, I can make head nor tails from reading the PR discussions. It appears to be closed and it's unclear what its status. There also appears to be some conflicts with other compilers. Please advise?

On another note, the official instructions for creating a recipe is like drinking from a firehose with bits of shrapnel in the stream. It starts out intuitively but quickly turns into unintelligible set "of do this if you find this", "oh by the way, if this thingamabob gives your problem, then do that". It's still unclear as to why brew recipes require a full compilation. The documentation could at least explain this. The cutesy brewery terminology adds to the frustration and confusion by mocking the user with annoying hipsteristics. 

 Anyway, I will work through this and see how far I can go.

**Reply by SMillerDev** [reply]
> It's still unclear as to why brew recipes require a full compilation.

Because we're a package manager that provides software from source

**Comment by Neved4** [comment]
@Joebeazelman https://github.com/rami3l/homebrew-tap/blob/master/alire.rb
